pkg/multicluster/watcher.go

```diff
@@ -18,33 +19,24 @@ const (
 	LocalCluster = ""
 )
 
-// ClusterHandler is passed to RunClusterWatcher to handle select cluster events.
-type ClusterHandler interface {
-	// ClusterHandler is called when a new cluster is identified by a cluster watch.
-	// The provided context is cancelled when a cluster is removed, so any teardown behavior for removed clusters
-	// should take place when ctx is cancelled.
-	HandleAddCluster(ctx context.Context, cluster string, mgr manager.Manager) error
-}
-
 type clusterWatcher struct {
 	ctx      context.Context
 	handlers []ClusterHandler
-	managers ManagerSet
+	cancels  *cancelSet
 }
 
 // RunClusterWatcher initializes and runs a reconciler for KubeConfig secrets.
-// It starts and runs provided ClusterHandlers for the localManager, and maintains a set of
-// active cluster manager.Managers in mangers.
-func RunClusterWatcher(ctx context.Context, localManager manager.Manager, managers ManagerSet, handlers ...ClusterHandler) error {
+// It starts and runs ClusterHandlers for the localManager as if it were discovered by the watcher.
+func RunClusterWatcher(ctx context.Context, localManager manager.Manager, handlers ...ClusterHandler) error {
```

Comment:
nit: what we are referring to as `localManager` might not in fact be a local manager. this `mgr` simply belongs to the cluster where we expect secrets. maybe masterManager?